### PR TITLE
Mipmaps

### LINKED ISSUE #782: Store mipmaps
Store precalculated mipmaps in the custom texture format.

### LINKED ISSUE #783: Load mipmaps
Load mipmaps from our custom texture format rather than generating them when loading the texture.

### PR DESCRIPTION
**Pull request by Chainsawkitten**
Store and load mipmaps in the custom format.

Textures have to be reimported (this reimports textures already in master).

Fixes #782 
Fixes #783

### CODE REVIEW

**Comment by NinjaDanz3r** [comment]
SonarQube analysis reported 1 issue

Note: The following issues were found on lines that were not modified in the pull request. Because these issues can't be reported as line comments, they are summarized here:

1. ![MINOR][MINOR] AncientFungus:LGEPR: Cppcheck cannot find all the include files (use --check-config for details) [rule](http://localhost:9000/coding_rules#rule_key=cppcheck%3AmissingInclude)

[MINOR]: https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR'

File: src/Editor/GUI/Editors/TextureEditor.cpp [hunk 1]
```diff
@@ -20,7 +20,7 @@ void TextureEditor::Show() {
         if (ImGui::InputText("Name", name, 128, ImGuiInputTextFlags_EnterReturnsTrue)) {
             // Rename texture files.
             std::string path = Hymn().GetPath() + "/" + texture->path;
-            rename((path + texture->name + ".png").c_str(), (path + name + ".png").c_str());
+            rename((path + texture->name + ".htc").c_str(), (path + name + ".htc").c_str());
```
Comment (NinjaDanz3r) [inline src/Editor/GUI/Editors/TextureEditor.cpp]: Is probably supposed to be ".hct"

**Comment by Chainsawkitten** [comment]
This should probably not be merged before the testing, since a lot of textures will have to be reimported.

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Comment by NinjaDanz3r** [comment]
iamheman

**Pull request merged**